### PR TITLE
Add pallet clearing house

### PR DESCRIPTION
**Pull request by 0xangelo**
This PR imports the Clearing House state as of https://github.com/ComposableFi/composable/pull/1487, its last approved PR.

A subsequent PR will bring the changes that I've made in the branch [`vamm-integration3`](https://github.com/ComposableFi/composable/tree/escher-feature/vamm-integration3).

### CODE REVIEW

File: clearing-house/src/tests/open_position.rs [hunk 1]
```diff
@@ -16,14 +16,14 @@ use crate::{
         any_direction, any_price, as_balance, get_collateral, get_market, get_market_fee_pool,
         get_outstanding_profits, get_position, run_for_seconds, run_to_time, set_fee_pool_depth,
         set_maximum_oracle_mark_divergence, set_oracle_price, set_oracle_twap,
-        with_markets_context, with_trading_context, Market, MarketConfig,
+        with_markets_context, with_trading_context, Market, MarketConfig, comp::{approx_eq_lower, approx_eq},
```
Comment (Cardosaum) [inline clearing-house/src/tests/open_position.rs]: Just a NIT: maybe there would be a better name for the file `comp.rs`? I don't think it's obvious that `comp` is short for `comparison` (it initially gave me the idea of 'computation'). Maybe a naming along the lines of `tolerance`, `approximations`, `check_diff` or `comparison` would be better?

Just an idea really, up to you to go with a change or not.

File: clearing-house/src/tests/open_position.rs [hunk 2]
```diff
@@ -555,42 +555,45 @@ proptest! {
...
+            let base_value_to_close = new_price
+                .try_mul(&base_amount_to_close.into_decimal().unwrap())
+                .unwrap()
+                .into_balance()
+                .unwrap();
+            let swapped = <TestPallet as ClearingHouse>::open_position(
+                &ALICE,
+                &market_id,
+                direction.opposite(),
+                dbg!(base_value_to_close),
```
Comment (Cardosaum) [inline clearing-house/src/tests/open_position.rs:570]: [Note to myself] 
Was that `dbg` removed in the later commits?

Comment (Cardosaum) [reply clearing-house/src/tests/open_position.rs:570]: Yeah, it seems it's still in the latest code this `dbg` call.
Should we keep it? 

cc/ @0xangelo

**Review by Cardosaum — APPROVED** [review]
LGTM! :rocket: 

Small comments, but for me it's good to merge :+1:

**Review by kevin-composable — APPROVED** [review]

**Comment by kevin-composable** [comment]
Looks great Angelo

File: clearing-house/src/tests/open_position.rs (hunk 2 above)
Comment (0xangelo) [reply clearing-house/src/tests/open_position.rs:570]: Good catch, I'll remove it next time.

**Pull request merged**